Tests/BasicTests/FileSystemTests.swift

```diff
@@ -134,6 +134,22 @@ class FileSystemTests: XCTestCase {
         XCTAssertTrue(fs.isFile(filePath))
         let data = try! fs.readFileContents(filePath)
         XCTAssertEqual(data, ByteString(testData))
+        
+        // Atomic writes
+        // In-memory file system doesn't have atomic writes implemented, so it throws.
+        XCTAssertThrows(FileSystemError.unsupported, { try Basic.InMemoryFileSystem(files: [:]).writeFileContents(filePath, bytes: ByteString(testData), atomically: true) })
```

Comment:
With the in-memory file system it should trivial to implement atomic writes, even without going through the temporary file. I think if you write as non-atomic, it'd appear atomic.

Reply:
Done, it's now treating the base implementation of in-memory file system's writeFileContents as atomic.